crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -83,8 +85,13 @@ impl RuffSettings {
     /// Constructs [`RuffSettings`] by merging the editor-defined settings with the
     /// default configuration.
     fn editor_only(editor_settings: &ResolvedEditorSettings, root: &Path) -> RuffSettings {
+        let mut config = Configuration::default();
+        if let Some(fallback_version) = find_fallback_target_version(root) {
+            config.target_version = Some(PythonVersion::from(fallback_version));
+        };
+
```

Comment:
Does that mean that we'll pick up the `requires-python` constraint even if a user uses `editor-only` in their settings. I'd be a bit confused if `editor-only` loads any project settings. @dhruvmanila what's your take on this?

Reply:
Yeah, I don't think we should be looking at any of the config files if the configuration preference is `editorOnly` unless the user has explicitly asks us using the `ruff.configuration`.

Reply:
That branch will be calling `RuffSettings::editor_only` to create a default configuration so we'll be getting there in this case.